app/controllers/internal/users_controller.rb

```diff
@@ -40,53 +42,62 @@ def update
     end
   end
 
-  def handle_mentorship
-    if user_params[:ban_from_mentorship] == "1"
-      ban_from_mentorship
+  def warn_or_ban_user
+    if user_params[:ban_user] == "1"
+      @user.add_role :banned
+      create_note("banned", user_params[:note_for_current_role])
+    elsif user_params[:warn_user] == "1"
+      @user.add_role :warned
+      create_note("warned", user_params[:note_for_current_role])
+    elsif user_params[:good_standing_user] == "1"
+      @user.remove_role :warned
+      create_note("good_standing", user_params[:note_for_current_role])
     end
+  end
 
-    if @new_mentee.blank? && @new_mentor.blank?
-      return
-    end
+  def add_note
+    return unless !user_params[:note].blank?
 
-    make_matches
+    create_note("misc_note", user_params[:note])
+  end
+
+  def create_note(reason, content)
+    Note.create(
+      author_id: @current_user.id,
```

Comment:
Guessing this is a typo and you meant `current_user` and not `@current_user` right?

```suggestion
      author_id: current_user.id,
```